internal/batches/ui/task_exec_tui.go

```diff
@@ -0,0 +1,391 @@
+package ui
```

Comment:
This is the reworked version of `batchProgressPrinter`. Much simplified but there's still a lot left (I didn't touch the whole diff stuff at the end, for example)